### PR TITLE
Fix feature image upload

### PR DESCRIPTION
**Pull request by orabina**
@gregory-claeyssens  work in progress, but this is what i am testing with and seems to work just fine 

Update the image uploader class to return the image ID to use it in `saveFeaturedImage` instead of getting the post id from the url.

### CODE REVIEW

File: includes/class.imageuploader.php [hunk 1]
```diff
@@ -95,22 +102,28 @@ public function save() {
         // check if file with same name exists in upload path
         if (is_file($image_path)) {
             $this->url = $image_url;
+            // If the image already exists, get the id to link it to the post
```
Comment (orabina) [inline includes/class.imageuploader.php:105]: When an image exists ( used before in another post or side-loaded when the first image in content is used as featured ) we need to find it and get the id

Comment (orabina) [reply includes/class.imageuploader.php:105]: There might be another way to get the id in this case, i am still looking...

Comment (orabina) [reply includes/class.imageuploader.php:105]: @gregory-claeyssens  I could not find a better way to do this
its kind of an edge case when an image is already used before and now used as a featured image,
the only other option is: do what was commented out a bit below ( if image path is used, rename and upload a new one ) 
what do you think

File: includes/class.imageuploader.php [hunk 2]
```diff
@@ -95,22 +102,18 @@ public function save() {
         // check if file with same name exists in upload path
         if (is_file($image_path)) {
             $this->url = $image_url;
-            curl_close($ch);
-            return true;
+            // If the image already exists, get the id to link it to the post
+            $attachment_id = attachment_url_to_postid($image_url);
```
Comment (orabina) [inline includes/class.imageuploader.php:106]: @gregory-claeyssens  this is the change based on our chat yesterday

Comment (gregory-claeyssens) [reply includes/class.imageuploader.php:106]: If it works, this is looking food imo :)

Comment (orabina) [reply includes/class.imageuploader.php:106]: Looking food ;) ⬆️ 

yes, so testing **with out** this change : 
featured image is not attached in wp in some cases ( already exists or already downloaded)

Testing **with** this change: 
Featured image seems to always be there in all my tests

**Review by gregory-claeyssens — APPROVED** [review]

**Pull request merged**